src/main/java/org/adridadou/ethereum/propeller/rpc/Web3JFacade.java

```diff
@@ -81,10 +81,21 @@ BigInteger getTransactionCount(EthAddress address) {
                     EthBlock currentBlock = web3j
                             .ethGetBlockByNumber(DefaultBlockParameter.valueOf(DefaultBlockParameterName.LATEST.name()), true).send();
                     BigInteger currentBlockNumber = currentBlock.getBlock().getNumber();
-                    if (this.lastBlockNumber.equals(BigInteger.ZERO) || currentBlockNumber.compareTo(this.lastBlockNumber) > 0) {
-                        this.lastBlockNumber = currentBlockNumber;
-                        blockEventHandler.newElement(currentBlock);
+
+                    //Set last block to current block -1 in case last block is zero to prevent all blocks from being retrieved
+                    if (this.lastBlockNumber.equals(BigInteger.ZERO)) {
+                        this.lastBlockNumber = currentBlockNumber.subtract(BigInteger.ONE);
+                    }
+
+                    //In case the block number of the current block is more than 1 higher than the last block, retrieve intermediate blocks
+                    for (BigInteger i = this.lastBlockNumber.add(BigInteger.ONE); i.compareTo(currentBlockNumber) < 0; i = i.add(BigInteger.ONE)) {
+                        EthBlock missedBlock = web3j.ethGetBlockByNumber(DefaultBlockParameter.valueOf(i), true).send();
+                        this.lastBlockNumber = i;
+                        blockEventHandler.newElement(missedBlock);
                     }
+
+                    this.lastBlockNumber = currentBlockNumber;
```

Comment:
do you need to reset it here? I would imagine that this.lastBlockNumber = i keeps it up to date no?

Reply:
The for-loop only retrieves the blocks between the lastBlockNumber and currentBlockNumber, and thus is skipped when there are no blocks in between. the currentBlock is not retrieved in this for-loop since we already retrieved it at the start of the while-loop.